python/ray/autoscaler/command_runner.py

```diff
@@ -682,3 +686,38 @@ def _docker_expand_user(self, string, any_char=False):
                 return string.replace("~", self.home_dir, 1)
 
         return string
+
+    def run_init(self, as_head):
+        image = self.docker_config.get("image")
+        if image is None:
+            image = self.docker_config.get(
+                "{}_image".format("head" if as_head else "worker"))
+
+        self._check_docker_installed()
+        if self.docker_config.get("pull_before_run", True):
+            assert image, "Image must be included in config if "
+            "pull_before_run is specified"
+
+            self.run("docker pull {}".format(image), run_env="host")
+
+        start_command = self.docker_config["{}_docker_start".format(
+            "head" if as_head else "worker")]
+
+        if not self._check_container_status():
+            self.run(start_command, run_env="host")
+        else:
+            running_image = self.run(
+                check_docker_image(self.container_name),
+                with_output=True,
+                run_env="host").decode("utf-8").strip()
+            if running_image != image:
+                logger.error(
+                    "A container with name {} ".format(self.container_name) +
+                    "is running image {} instead ".format(running_image) +
+                    "of {} (which was provided in the YAML".format(image))
```

Comment:
we can use fstrings now :) 
```suggestion
                    "A container with name {} ".format(self.container_name) +
                    "is running image {} instead ".format(running_image) +
                    "of {} (which was provided in the YAML).".format(image))
```

Reply:
Oh yep! Excited for all the fstrings!